pkg/reconciler/managed/reconciler.go

```diff
@@ -552,7 +616,7 @@ func NewReconciler(m manager.Manager, of resource.ManagedKind, o ...ReconcilerOp
 }
 
 // Reconcile a managed resource with an external resource.
-func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconcile.Result, error) { // nolint:gocyclo
+func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (result reconcile.Result, err error) { // nolint:gocyclo
```

Comment:
I'd like to avoid named return values here - this function is very complex and I think it's likely to become a footgun.

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters